SFS_ir/interpolate_ir.m

```diff
@@ -122,6 +124,43 @@
         ir = ifft(magnitude.*exp(1i*phase),[],3);
         % Avoid round-off errors
         ir = real(ir);
+    case 'timedomain'
+        % See Hartung et al. (1999)
+        %
+        % Loop over channels
+        ir_results = zeros(1,size(ir,2),size(ir,3));
+        for n = 1:size(ir,2)
+            % Determine TOA differences
+            ir_upsampled = resample(squeeze(ir(:,n,:))',10,1);
+            for nn = 1:size(ir,1)
+                [~,idx_max(nn)] = ...
+                    max(xcorr(ir_upsampled(:,1),ir_upsampled(:,nn)));
+            end
+            N = size(ir_upsampled,1);
+            k = -(N-1):(N-1);
+            shifts = k(idx_max);
+            
+            % Shift all irs back to align with last ir
+            max_shift = max(shifts)-min(shifts);
+            TOA_diff_to_last = shifts-min(shifts);
+            ir_shifted = zeros(N+max_shift,size(ir,1));
+            for nn = 1:size(ir,1)
+                ir_shifted(:,nn) = [zeros(TOA_diff_to_last(nn),1); ...
+                    ir_upsampled(:,nn); zeros(max_shift-TOA_diff_to_last(nn),1)];
+            end
+            
+            % Interpolate aligned irs
+            ir_interp = sum(bsxfun(@times,ir_shifted,weights'),2);
+            
+            % Interpolate TOA differences and shift interpolated ir accordingly
+            TOA_diff_interp = round(TOA_diff_to_last*weights);
+            ir_interp = ...
+                ir_interp(TOA_diff_interp+1:end-(max_shift-TOA_diff_interp));
```

Comment:
This is cropping the interpolated impulse response and thereby shifting further to beginning by `TOA_diff_interp`, correct? Why is this step needed?

Reply:
First all irs have been shifted further to the end to align them with the ir that arrives the latest by adding zeros at the beginning. The interpolate aligned ir then has to shifted back towards the beginning according to the interpolated TOA difference. This means samples at the start have to be cropped.
The last samples are cropped to preserve the original length of the ir.

Reply:
Ah, ok, now I got it.